### PR TITLE
Support Magisk v20.2 SEPolicy API and fix bootloop

### PR DESCRIPTION
**Pull request by MlgmXyysd**
Magisk v20.2 added a new API to add support for custom sepolicy rules to be applied in the pre-init phase [1,2].

This pull-request
1) Updates the Magisk installer script to the latest version [1]
2) Adds support for custom sepolicy rules [2]
3) Converts the install method to the new variant using `customize.sh` [3]
4) Fix OPPO Android P bootloop [5]
5) Fix SandHook npe [6]
6) Add stub app from https://github.com/ElderDrivers/EdXposedManager [7]

These rules are merged by Magisk and only applied once during the pre-init phase.

This should make the **custom Magisk build obsolete** (when using Magisk v20.2+).

[1] https://github.com/topjohnwu/Magisk/commit/12fda2928051601ebc7fac980fae29a80e7979cc
[2] https://topjohnwu.github.io/Magisk/guides.html#sepolicyrule
[3] https://topjohnwu.github.io/Magisk/guides.html#magisk-module-installer
[4] https://github.com/ElderDrivers/EdXposed/pull/428
[5] https://github.com/wuxianlin/EdXposed/commit/e009fa88572943232800cc337c7b568098d4c86d
[6] https://github.com/ganyao114/SandHook/pull/45
[7] https://github.com/ElderDrivers/EdXposedManager/commit/56547eb35bcdb7fd75a738adeea987cff0afc3a9

### Thanks
- @m0vie 
- @wuxianlin

### CODE REVIEW

File: edxp-core/template_override/sepolicy.rule [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # necessary for using mmap in system_server process
```
Comment (MlgmXyysd) [inline edxp-core/template_override/sepolicy.rule]: There should be no comments here

**Comment by WiredLife** [comment]
Is there a compiled version to install via magisk?
Thanks!

**Comment by MlgmXyysd** [comment]
@WiredLife https://t.me/EdXposed/43

**Comment by Mikanoshi** [comment]
When was EdXposed Installer support dropped?

**Comment by MlgmXyysd** [comment]
> When was EdXposed Installer support dropped?
> I also have the same error with 0.4.6.1_4497: #443
> Framework works though.

Installer support dropped at https://github.com/ElderDrivers/EdXposed/pull/431/commits/461bf5867a59c7f1efa9263233fcf18e0b8fe8b9

re-install edxposed manager

**Comment by Mikanoshi** [comment]
> re-install edxposed manager

Installed it 5 times) It's not the main problem though, EdXposed Manager ignores apk that I update from Android Studio via ADB almost all the time, it does not update module list, it does not show any notification, while Installer works fine on same setup.

**Comment by MlgmXyysd** [comment]
@Mikanoshi uninstall and install

**Comment by Mikanoshi** [comment]
@MlgmXyysd No, that's not helping. Can you just add a null check here? It won't break anything, thisObject is already null when error is thrown.
https://github.com/ElderDrivers/EdXposed/blob/master/edxp-common/src/main/java/com/elderdrivers/riru/edxp/hooker/XposedInstallerHooker.java#L31

**Comment by HpNoTiQ56** [comment]
What's included in Stub app and how to use it. I've compiled your last commit of Manager and I've got two APK : stub and release. Is it useful to pass S/N?

**Comment by MlgmXyysd** [comment]
@HpNoTiQ56 no, just a stub, for give a link to download EdXposed Manager, and prevent to use unofficial closed-source EdXposed Manager

**Comment by Mikanoshi** [comment]
Not if you have signature check removed :) Anyway, do you really need android support lib that makes apk almost 1MB just to show a dialog and open a link? Could be 50KB.

**Comment by nift4** [comment]
Why is this not merged? I use the canary buikds with this PR

**Comment by MlgmXyysd** [comment]
> Not if you have signature check disabled :) Anyway, do you really need an android support lib that makes apk almost 1MB just to show a dialog and open a link? Could be 50KB.

If I don't use appcompat, you'll only see Android 4. X style dialogs

**Comment by MlgmXyysd** [comment]
> Why is this not merged? I use the canary builds with this PR on my daily driver Android Q.

not completed

**Comment by nift4** [comment]
Oops

**Comment by MlgmXyysd** [comment]
TODO: 
[ ] Completed persist detection
[ ] Stub app
[ ] Module activation scope [1]

[1] [Talk about Xposed new concept <Modules Activation Scope> (Chinese) ](https://www.bilibili.com/read/cv4733304)

**Comment by MlgmXyysd** [comment]
> Not if you have signature check disabled :) Anyway, do you really need an android support lib that makes apk almost 1MB just to show a dialog and open a link? Could be 50KB.

Now removed appcompat

**Comment by MlgmXyysd** [comment]
*If no more issues about persist partition detection (or sepolicy.rule) are submitted, this PR will be merged on March 1, 2020.

**Pull request merged**